datafusion/common/src/dfschema.rs

```diff
@@ -384,8 +384,7 @@ impl DFSchema {
         let self_fields = self.fields().iter();
         let other_fields = other.fields().iter();
         self_fields.zip(other_fields).all(|(f1, f2)| {
-            f1.qualifier() == f2.qualifier()
-                && f1.name() == f2.name()
+            f1.qualified_name() == f2.qualified_name()
```

Comment:
About this, I have a new idea on weekend. we may need an `alias_with_schema()` function that allows us to not only add alias, but also specify the schema.

Because `alias('t1.a')`, `field` is `qualifier: none, name: t1.a`, we hope `field` is `qualifier: t1, name: a`.

I think `alias_with_schema()` can achieve this goal.

Reply:
related issue #6681